src/geo.rs

```diff
@@ -171,6 +185,14 @@ fn split_triangle(triangle: Triangle) -> [Triangle; 2] {
     ]
 }
 
+/// Returns the 3D point halfway between v0 and v1.
+fn midpoint_3d(v0: Vector<f32, 3>, v1: Vector<f32, 3>) -> Vector<f32, 3> {
+    let diff = v0 - v1;
+    v1 + (diff / 2.0)
```

Comment:
Isn't this the same as (v0 + v1)/2.0 ?

Reply:
Heh. You are absolutely correct. That's both easier to read and simpler. And the more standard way of computing it. Completely missed that. Thanks